src/main/java/ca/corefacility/bioinformatics/irida/ria/web/files/SequenceFileAjaxController.java

```diff
@@ -0,0 +1,117 @@
+package ca.corefacility.bioinformatics.irida.ria.web.files;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import ca.corefacility.bioinformatics.irida.exceptions.EntityNotFoundException;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.model.workflow.analysis.AnalysisFastQC;
+import ca.corefacility.bioinformatics.irida.ria.web.files.dto.FastQCDetailsResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.files.dto.FastQCImagesResponse;
+import ca.corefacility.bioinformatics.irida.service.AnalysisService;
+import ca.corefacility.bioinformatics.irida.service.SequencingObjectService;
+
+/**
+ * Ajax controller to get data for the fastqc page.
+ */
+@RestController
+@Scope("session")
+@RequestMapping("/ajax/sequenceFiles")
+public class SequenceFileAjaxController {
+
+	private AnalysisService analysisService;
+	private SequencingObjectService sequencingObjectService;
+
+	public static final String IMG_PERBASE = "perbase";
+	public static final String IMG_PERSEQUENCE = "persequence";
+	public static final String IMG_DUPLICATION_LEVEL = "duplicationlevel";
+
+	public static final String[] qcFileTypes = { "perbase", "persequence", "duplicationlevel" };
+
+	@Autowired
+	public SequenceFileAjaxController(AnalysisService analysisService,
+			SequencingObjectService sequencingObjectService) {
```

Comment:
Why are you not creating a UI service to handle this stuff?

Reply:
Added in 8d915d2b5135ef05b2f4eafaebe586d755b13903